src/js/component.js

```diff
@@ -38,14 +38,16 @@ import window from 'global/window';
  * @constructor
  * @extends vjs.CoreObject
  */
-var Component = CoreObject.extend({
-  /**
-   * the constructor function for the class
-   *
-   * @constructor
-   */
-  init: function(player, options, ready){
-    this.player_ = player;
+class Component {
+
+  constructor(player, options, ready){
+
+    // The component might be the player itself and we can't pass `this` to super
+    if (!player && this.play) {
+      this.player_ = player = this;
```

Comment:
Does the player not get the player argument?
I wonder if we can use default args here:
```js
class Component {
  constructor (player = this, options, ready) {
    this.player_ = player;
  }
}
```